api/parse.py

```diff
@@ -32,111 +33,101 @@ def food_database():
 # present in our output code
-def patter_match():
-
-    nlp = spacy.load('en_core_web_sm')
+# returns the items present in the reciept as well as the 
+# count of each item in dict() format
+def pattern_match():
+    nlp = spacy.load('en')
     matcher = PhraseMatcher(nlp.vocab, attr = 'LOWER')
-
-   # parse the reciept words and covert the list to a string 
+    
     parsed_words = parse()
+    
+    '''
+    parsed_words = ['walmart', 'save money. live better.', 
+    '( 330 ) 339 -', '3991', 'manager diana earnest',
+     '231 bluebell dr sw', 'new philadelphia oh 44663',
+      '02115 009044 44', '01301', 'pet toy', 'floppy puppy', 
+      'sssupreme s', 'z . 5 squeak', 'munchy dmbel', 'dog treat', 
+      'ped pch 1', 'ped pch 1', 'coupon 23100', '1--inymd smores',
+       'french drsng', '3 oranges', 'baby carrots', 'collards',
+        'calzone', 'mm rvw mnt', 'stkobrlplabl', 'stkobrlplabl', 
+        'stko sunflwr', 'stko sunflwr', 'stko sunflwr',
+        'stko sunflwr', 'bling beads', 'great value', 
+        'lipton', 'dry dog', 'tax', 'us debit', '004747571658', 
+        '004747514846', '070060332153', '084699803238', '068113108796',
+         '007119013654', '002310011802', '002310011802', '052310037000', '088491226837', '004132100655', '001466835001', '003338366602', '1', '000000004614k1', '005208362080 f', '003399105848', '001558679414', '001558679414', '001558679410', '001558679410', '001558679410', '001558679410', '076594060699', '007874203191 f', '001200011224 f', '002310011035', '1', 'subtotal', '6.750', 'total', 'visa tend', '5', '12 .', '9166', '1.97', '1.97', '4.97', '5.92', '3.77', '2.92', '0.50', '0.50', '1.00-0', '3.98', '1.98', '.47', '1.48', '1.24', '2.50', '19.77', '1.97', '1.97', '0.97', '0.97', '0.97', '0.97', '0.97', '9.97', '4.48', '44', '93.62', '4.59', '98.21', '98.21', '1', 'x', 'x', 'x', 'x', 'x', 'x', 'x', 'x', 'o', 'o', 'o', 'x', 'x', 'x', 'x', 'x', 'x', 'x', 'x', 'o', 'x', 'x', 'o', 'approval # 572868', 'ref # 720900544961', 'trans id', '387209239650894', 'validation - 87hs', 'payment service', 'aid a0000000980840', 'tc 51319ca81dcz2bc7', 'terminal # sc010764', '*signature ver ified', '07/28/17', '02 : 39 : 48', 'change due', '# items sold 25', '0443 0223 1059 8001 5140', 'low'
+    , 'ices you can trust .', 'every', '0.00', 'day .', '07/28/17', 
+    '02 : 39 : 48']
+    ''' 
+    
     words_string = ' '.join(parsed_words)
+    (single,plural) = filter_words(nlp,words_string)
+    
+    
+    text_doc_singular = nlp(single)
+    text_doc_plural   = nlp(plural)
     food_groups = food_database()
-    fruit_veg_string = food_groups[3]]
 
     # Building the intial list of keywords we want to match the parsed items aganist
     patterns = [nlp(text) for text in food_groups]
-    matcher.add("TerminologyList", None, *patterns)
-
-    text_doc = nlp(words_string)
-    matches = matcher(text_doc)
+    matcher.add("Food Matcher", None, *patterns)    
+    matches = matcher(text_doc_singular)
```

Comment:
What about `matches(text_doc_plural)`?

Reply:
it's there after the for loop for singular matching